sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2648,18 +2648,26 @@ case class StringDecode(bin: Expression, charset: Expression)
 
   protected override def nullSafeEval(input1: Any, input2: Any): Any = {
     val fromCharset = input2.asInstanceOf[UTF8String].toString
-    UTF8String.fromString(new String(input1.asInstanceOf[Array[Byte]], fromCharset))
+    try {
+      UTF8String.fromString(new String(input1.asInstanceOf[Array[Byte]], fromCharset))
```

Comment:
since we are here, can we rewrite this expression using `StaticInvoke`? To avoid this problem completely.

Reply:
The same to `Encode`

Reply:
Shall we introduce the `legacyCharsets` and `supportedCharsets` too?